src/benchmark/operators/sort_benchmark.cpp

```diff
@@ -1,22 +1,181 @@
+#include <expression/expression_functional.hpp>
 #include <memory>
 
 #include "benchmark/benchmark.h"
 
 #include "../micro_benchmark_basic_fixture.hpp"
+#include "operators/limit.hpp"
 #include "operators/sort.hpp"
 #include "operators/table_wrapper.hpp"
+#include "synthetic_table_generator.hpp"
+#include "operators/join_hash.hpp"
 
 namespace opossum {
 
-BENCHMARK_F(MicroBenchmarkBasicFixture, BM_Sort)(benchmark::State& state) {
-  _clear_cache();
+class SortBenchmark : public MicroBenchmarkBasicFixture {
+ public:
+  void BM_Sort(benchmark::State& state) {
+    _clear_cache();
 
-  auto warm_up = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
-  warm_up->execute();
-  for (auto _ : state) {
-    auto sort = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
-    sort->execute();
+    auto warm_up = std::make_shared<Sort>(_table_wrapper_a, ColumnID{0} /* "a" */, OrderByMode::Ascending);
+    warm_up->execute();
```

Comment:
I know that this is a best practice. Still wondering: Did you see an actual difference with/out warmup?

Reply:
These are the results without the `warm_up`:
```
Bastian.Koenig@vm-appleton:~/hyrise$ cmake-build-release/hyriseMicroBenchmarks --benchmark_filter=BM_Sort*
2019-12-16 17:12:49
Running cmake-build-release/hyriseMicroBenchmarks
Run on (32 X 2500 MHz CPU s)
CPU Caches:
  L1 Data 32K (x32)
  L1 Instruction 32K (x32)
  L2 Unified 256K (x32)
  L3 Unified 25600K (x32)
Load Average: 0.29, 0.14, 0.05
--------------------------------------------------------------------------------------------
Benchmark                                                  Time             CPU   Iterations
--------------------------------------------------------------------------------------------
SortBenchmark/BM_Sort                               10013751 ns     10013638 ns           84
SortPicoBenchmark/BM_SortPico                           3115 ns         3114 ns       213037
SortSmallBenchmark/BM_SortSmall                       701094 ns       701036 ns          746
SortLargeBenchmark/BM_SortLarge                     88207271 ns     88196949 ns            9
SortReferenceBenchmark/BM_SortReference             15162234 ns     15161228 ns           56
SortReferencePicoBenchmark/BM_SortReferencePico         3709 ns         3708 ns       186817
SortReferenceSmallBenchmark/BM_SortReferenceSmall     902671 ns       902670 ns          594
SortReferenceLargeBenchmark/BM_SortReferenceLarge  173210621 ns    173210489 ns            3
SortStringBenchmark/BM_SortString                   32152662 ns     32148606 ns           27
SortStringSmallBenchmark/BM_SortStringSmall          3176690 ns      3176126 ns          179
SortStringLargeBenchmark/BM_SortStringLarge        516888142 ns    516836706 ns            2
SortNullBenchmark/BM_SortNullBenchmark              14963719 ns     14963718 ns           57
```

Reply:
The results show: The microbenchmarks indeed perform noticeably differently with and without warm-up – although we're not really sure about which way is “better” (see e.g. the SortNullBenchmark or SortStringLarge, where the warm-up _increased_ the amount of iterations with disabled warm-up instead of _decreasing_ it as with most of the Benchmarks), we think that leaving the warm-up enabled is good at this point.